src/controllers/csi/provisioner/agent.go

```diff
@@ -42,71 +49,95 @@ func newAgentUpdater(
 	return &agentUpdater{
 		fs:        fs,
 		path:      path,
+		apiReader: apiReader,
 		recorder:  recorder,
 		dk:        dk,
 		installer: agentInstaller,
 	}
 }
 
-func (updater *agentUpdater) updateAgent(installedVersion, tenantUUID string, previousHash string, latestProcessModuleConfigCache *processModuleConfigCache) (string, error) {
+func (updater *agentUpdater) updateAgent(ctx context.Context, latestVersion, tenantUUID string, previousHash string, latestProcessModuleConfigCache *processModuleConfigCache) (string, error) {
```

Comment:
Move the context and latestVersion into the agentUpdater constructor. The previousHash string is not used anymore in this function. latestVersion is also only used for logging, maybe the logging could be done somewhere else or in different way so this parameter does not have to be passed as well